native/src/App.tsx

```diff
@@ -36,17 +40,37 @@ const linking: LinkingOptions = {
       [REDIRECT_ROUTE]: '*'
     }
   },
-  getStateFromPath: path => {
-    return {
-      index: 0,
-      routes: [
-        {
-          name: REDIRECT_ROUTE,
-          params: {
-            url: `https://${path}`
-          }
+  getStateFromPath: path => ({
+    index: 0,
+    routes: [
+      {
+        name: REDIRECT_ROUTE,
+        params: {
+          url: `https://${path}`
         }
-      ]
+      }
+    ]
+  }),
+  subscribe: (listener: (url: string) => void) => {
+    const onReceiveURL = ({ url }: { url: string }) => listener(url)
+
+    Linking.addEventListener('url', onReceiveURL)
+
+    const unsubscribeNotification = messaging().onNotificationOpenedApp(() => {
+      console.log('on notification opened app')
+      listener(
+        urlFromRouteInformation({
+          cityCode: CITY_CODE_PLACEHOLDER,
```

Comment:
It is now possible to open deep links of the following style: `https://integreat.app/city_code_placeholder/language_code/placeholder/news/local` without specifying the city and language.

Reply:
But that doesn't really sound like a long term solution, right? What about people who switch cities. Do they receive push notifications for both cities or only the one which is currently selected?

Reply:
Upon switching cities the channel of the old city is unsubscribed, so they should only receive PNs of the currently selected city. And yes, there is also [this issue](https://github.com/Integreat/integreat-cms/issues/760) in the cms and [this issue](https://issues.tuerantuer.org/browse/IGAPP-263) in our project to change this. Still, I don't like my current solution either.

Reply:
and we don't have to merge this

Reply:
I had a look on this PR some time ago and did not know what to feedback you and I still feed the same way. Not sure what to do with this...

Reply:
I think generally the requirement of opening a link within the current city/language without specifying it is fine. This can be implemented in a proper was I think.

The major design issue is that we never can be sure that the notification which is shown actually belongs to the current city. For example if I receive a PN, then switch the city, and then click on the Notification. What should happen?

If we allow to open other cities via a PN then we really require changes in the CMS. If we do not want to allow this then the current CMS implementation is enough.

Reply:
Yes, it is already planned that the cms delivers city code, language code and news id in the push notifications, but this will only happen with django.

Reply:
Removed placeholders completely as we can just load city and language from async storage :facepalm: